sdks/java/io/iceberg/src/main/java/org/apache/beam/sdk/io/iceberg/ScanTaskReader.java

```diff
@@ -190,7 +192,8 @@ public Row getCurrent() throws NoSuchElementException {
     if (current == null) {
       throw new NoSuchElementException();
     }
-    return IcebergUtils.icebergRecordToBeamRow(source.getSchema(), current);
+    return icebergRecordToBeamRow(
+        icebergSchemaToBeamSchema(source.getSchema()), checkStateNotNull(current));
```

Comment:
The schema conversion may not be as light as you want it for a per-element operation. Can you cache the result of `icebergSchemaToBeamSchema`?

Reply:
Oh wow great catch, that def would've eaten up a good chunk of CPU usage

Reply:
Done